src/store/postSlice.js

```diff
@@ -65,6 +66,34 @@ export const deletePost = createAsyncThunk(
   }
 );
 
+export const likePost = createAsyncThunk(
+  "post/likePost",
+  async (postId, { rejectWithValue }) => {
+    try {
+      const res = await POST(`${API.LIKE_POST}/${postId}`, {});
+      if (res?.status === 200 || res?.status === 201) {
+        return res?.data.posts;
+      }
+    } catch (err) {
+      rejectWithValue(err.message);
+    }
+  }
+);
+
+export const dislikePost = createAsyncThunk(
+  "post/dislikePost",
+  async (postId, { rejectWithValue }) => {
+    try {
+      const res = await POST(`${API.DISLIKE_POST}/${postId}`, {});
+      if (res?.status === 200 || res?.status === 201) {
+        return res?.data.posts;
+      }
+    } catch (err) {
+      rejectWithValue(err.message);
```

Comment:
I got to know about this method rejectWithValue()
I will be using in my project for error handling